### PR TITLE
Embedded images in messages

### PR DESCRIPTION
**Pull request by stkem**
@joonho1101 @2is10 @andrewconner

### CODE REVIEW

**Comment by joonhocho** [comment]
Andrew, Stephen and I discussed and we are aware of that this is not very safe way to embed an image.
Stephen says that eventually server should generate a preview for every link and send a link to the preview to the client.

File: packrat/scripts/formatting.js [hunk 1]
```diff
@@ -16,7 +16,11 @@ function getTextFormatter() {
       var bits = parts[i].split(/(?:\b|^)((?:(?:(https?|ftp):\/\/|www\d{0,3}[.])?(?:[a-z0-9](?:[-a-z0-9]*[a-z0-9])?\.)+(?:com|edu|biz|gov|in(?:t|fo)|mil|net|org|name|coop|aero|museum|[a-z][a-z]\b))(?::[0-9]{1,5})?(?:\/(?:[^\s()<>]*[^\s`!\[\]{};:.'",<>?«»()“”‘’]|\((?:[^\s()<>]+|(?:\([^\s()<>]+\)))*\))*|\b))(?=[\s`!()\[\]{};:.'",<>?«»“”‘’]|$)/);
       for (var j = 1; j < bits.length; j += 3) {
         var escapedUri = Mustache.escape(bits[j]);
-        bits[j] = '<a target=_blank href="' + (bits[j+1] ? ""  : "http://") + escapedUri + '">' + escapedUri + "</a>";
+        var imageRegex = /^([^?#])*(\.(jpg|gif|jpeg|png))$/i; //Not very good, but fairly ysafe
+        var isImage = imageRegex.test(bits[j]);
+        var imageTag = '<a class="kifi-previewable-link" target="_blank" href="' + (bits[j+1] ? ""  : "http://") + escapedUri + '">' + '<img style="max-width: 100%;" src="' + (bits[j+1] ? ""  : "http://") + escapedUri + '">' + "</a>";
+        var linkTag = '<a class="kifi-previewable-link" target="_blank" href="' + (bits[j+1] ? ""  : "http://") + escapedUri + '">' + escapedUri + "</a>";
```
Comment (2is10) [inline packrat/scripts/formatting.js:22]: What is the significance of the class `kifi-previewable-link`? Why is it added to both text links and image links?

**Comment by 2is10** [comment]
It sounds like we’ve decided to go ahead with this as an experimental/prototype feature for now, so I’ll merge.

File: packrat/scripts/formatting.js (hunk 1 above)
Comment (stkem) [inline packrat/scripts/formatting.js:22]: My bad. I had originally intended to make it preview in a qtip on hover, but changed it after talking with Joon and Andrew and forgot to remove the class.

**Pull request merged**